pkg/collector/corechecks/snmp/internal/report/report_bandwidth_usage.go

```diff
@@ -96,3 +105,67 @@ func (ms *MetricSender) sendBandwidthUsageMetric(symbol checkconfig.SymbolConfig
 	ms.sendMetric(sample)
 	return nil
 }
+
+func (ms *MetricSender) sendIfSpeedMetrics(symbol checkconfig.SymbolConfig, fullIndex string, values *valuestore.ResultValueStore, tags []string) {
+	// We are piggybacking on presence of ifHCInOctets as a way to 1/ submit ifSpeed metrics only once, 2/ have corresponding fullIndex and 3/ tags.
+	// If needed, we can improve (at cost of complexity) by sending ifSpeed metrics based on presence of multiple metrics like ifHCInOctets/ifHCOutOctets/ifHighSpeed.
+	// I think it's reasonable for now, for simplify, to only rely on ifHCInOctets that should be present in the vast majority of cases.
+	if symbol.Name != "ifHCInOctets" {
+		return
+	}
```

Comment:
Am I right in saying what's happening here is that we're taking the data from `ifHCInOctets` which should be present in the vast majority of devices to get the tags, full index, etc. when trying to send the custom values? Thus, if `ifHCInOctets` is absent we won't attempt to send `ifInSpeed` or `ifOutSpeed`?

Reply:
@ken-schneider This is exactly that. We need to piggyback on one me of metrics for interface tags.

We can possibly rely on the presence of multiple data e.g. ifHCInOctets/ifHCOutOctets/ifHighSpeed, but the drawback is more complex code.
I think relying on the presence of `ifHCInOctets` should be good enough for now and we can still use more complex logic if needed later.